File: 2022/day1.py
import time

INPUT_FILE = "day1.txt"

def check_values_list(lst, value):

    for index, element in enumerate(lst):
        if value > element:
            lst.insert(index, value)
            lst.pop()
            break

def main():
    count = 0
    prev = 0
    number = 0
    values_list = [0, 0, 0]

    with open(INPUT_FILE, 'r') as file:
        for line in file:
            value = line.strip()
            try:
                number = int(value)
                count += number

            except ValueError:
                check_values_list(values_list, count)

                count = 0
                continue
 
    check_values_list(values_list, count)
    print(f"Part 1 = {values_list[0]}")
    print(f"Part 2 = {sum(values_list)}")

main()

File: 2022/day1.swift
import Foundation

let INPUT_FILE: String = "details/day1.txt"

func check_lst(lst: inout [Int], count: Int)
{
    
    for (index, element) in lst.enumerated()
    {
        if count > element
        {
            lst.insert(count, at: index)
            lst.removeLast()
            break
        }
    }

    return
}

func main() 
{
    var count: Int = 0
    var lst: [Int] = [0, 0, 0]

    do 
    {
        let contents = try String(contentsOfFile: INPUT_FILE, encoding: .utf8)
        for line in contents.split(separator: "\n", omittingEmptySubsequences: false) 
        {
            let trimmed = line.trimmingCharacters(in: .whitespacesAndNewlines)
            if trimmed.isEmpty 
            {
                check_lst(lst: &lst, count: count)
                count = 0
                continue
            }

            if let number = Int(trimmed) 
            {
                count += number
            }
        }
    } 

    catch 
    {
        print("Error: \(error.localizedDescription)")
    }
    
    let total = lst.reduce(0 , +)

    print("Part 1 = \(lst[0])")
    print("Part 2 = \(total)")
}

main()
